From 939a6352df8906952fd1f24276eeeb499fa3d2f4 Mon Sep 17 00:00:00 2001
From: Luiz Angelo Daros de Luca <luizluca@gmail.com>
Date: Tue, 9 Jan 2024 00:12:24 -0300
Subject: [PATCH 01/12] dt-bindings: net: dsa: realtek: fix user mdio for mdio
 switches

The mdio node is required for SMI or MDIO switches.

Also add the missing phy-handle for each port pointing to the now
present mdio ethernet-phy.

Signed-off-by: Luiz Angelo Daros de Luca <luizluca@gmail.com>
---
 .../devicetree/bindings/net/dsa/realtek.yaml  | 40 +++++++++++++++++--
 1 file changed, 37 insertions(+), 3 deletions(-)

--- a/Documentation/devicetree/bindings/net/dsa/realtek.yaml
+++ b/Documentation/devicetree/bindings/net/dsa/realtek.yaml
@@ -118,21 +118,19 @@ then:
     required:
       - mdc-gpios
       - mdio-gpios
-      - mdio
 
   properties:
     mdc-gpios: false
     mdio-gpios: false
-    mdio: false
 
 else:
   required:
     - mdc-gpios
     - mdio-gpios
-    - mdio
 
 required:
   - compatible
+  - mdio
 
     #  - mdc-gpios
     #  - mdio-gpios
@@ -351,26 +349,31 @@ examples:
                             port@0 {
                                     reg = <0>;
                                     label = "lan4";
+                                    phy-handle = <&eth3phy0>;
                             };
 
                             port@1 {
                                     reg = <1>;
                                     label = "lan3";
+                                    phy-handle = <&eth3phy1>;
                             };
 
                             port@2 {
                                     reg = <2>;
                                     label = "lan2";
+                                    phy-handle = <&eth3phy2>;
                             };
 
                             port@3 {
                                     reg = <3>;
                                     label = "lan1";
+                                    phy-handle = <&eth3phy3>;
                             };
 
                             port@4 {
                                     reg = <4>;
                                     label = "wan";
+                                    phy-handle = <&eth3phy4>;
                             };
 
                             port@7 {
@@ -386,5 +389,36 @@ examples:
                                     };
                             };
                     };
+
+                    mdio {
+                            #address-cells = <1>;
+                            #size-cells = <0>;
+
+                            eth3phy0: ethernet-phy@0 {
+                                    reg = <0>;
+                                    interrupt-parent = <&switch_intc2>;
+                                    interrupts = <0>;
+                            };
+                            eth3phy1: ethernet-phy@1 {
+                                    reg = <1>;
+                                    interrupt-parent = <&switch_intc2>;
+                                    interrupts = <1>;
+                            };
+                            eth3phy2: ethernet-phy@2 {
+                                    reg = <2>;
+                                    interrupt-parent = <&switch_intc2>;
+                                    interrupts = <2>;
+                            };
+                            eth3phy3: ethernet-phy@3 {
+                                    reg = <3>;
+                                    interrupt-parent = <&switch_intc2>;
+                                    interrupts = <3>;
+                            };
+                            eth3phy4: ethernet-phy@4 {
+                                    reg = <4>;
+                                    interrupt-parent = <&switch_intc2>;
+                                    interrupts = <4>;
+                            };
+                    };
             };
       };
